docs/source/design.rst

```diff
@@ -1,7 +1,151 @@
-.. _internals:
+.. _design:
 
-*********************
-Ibis design internals
-*********************
+Design
+======
 
-More to come here.
+Primary Goals
+*************
+
+#. Type safety
+#. Expressiveness
+#. Composability
+#. Familiarty
+
+Flow of execution
```

Comment:
maybe should capitalize all works in sections (or not), just do this consistently

Reply:
Done